### PR TITLE
chore(frontend): Real Tailwind + Autoprefixer

### PR DESCRIPTION
**Pull request by Twixes**
## Problem

We were faking Tailwind, but this was getting annoying. I wanted to use container queries just now to fix empty states banners being squished, and container queries _are_ supported by Tailwind – but would just be another big, mostly wasteful expansion of our contraption if we added that too.
Also we didn't actually use Autoprefixer despite having it configured, because it wasn't include in our ESBuild pipeline.

## Changes

Adds actual Tailwind! There's no trick here, it seems to work really well.
Initial frontend compilation takes a bit longer at ~12s instead of ~8s, but that's only incurred when starting the frontend, and not felt much. Subsequent rebuilds due to component updates are unaffected, still around 3s – although that seems to be because Tailwind processing does _not_ run on rebuilds, thus if you use a utility class that wasn't in use anywhere before, you'll need to do a full restart of the development frontend.

Added Autoprefixer as a bonus.

## How did you test this code?

Let's see what visual regression tests say.

### CODE REVIEW

**Review by benjackwhite — COMMENTED** [review]
Omg I want this to work so bad 🤞🤞
Will give it a proper review when I'm back from lunch...

**Comment by thmsobrmlr** [comment]
You're my hero for the next month, if this works!! 🥳

**Review by pauldambra — COMMENTED** [review]
💖

File: frontend/src/styles/global.scss [hunk 1]
```diff
@@ -15,7 +15,10 @@ Only 400 (`normal`), 500 (`var(--font-medium)`), 600 (`var(--font-semibold)`), o
 @import 'functions';
 @import 'mixins';
 @import 'utility-legacy';
-@import 'utilities';
+@import 'colors';
+@tailwind base;
```
Comment (benjackwhite) [inline frontend/src/styles/global.scss:19]: I believe this is overriding some global styles that we have previously relied on from Antd. 

In particular standard headings have some styles applied that seem to be reset in the tailwind base.
Generally I think this is a good thing, as the inherited antd styles are pretty obscure, we just need to add our own here.

Screenshot 2023-12-06 at 09 31 33

File: frontend/src/styles/global.scss [hunk 2]
```diff
@@ -15,7 +15,10 @@ Only 400 (`normal`), 500 (`var(--font-medium)`), 600 (`var(--font-semibold)`), o
...
+@tailwind components;
```
Comment (benjackwhite) [inline frontend/src/styles/global.scss:20]: Web Analytics is pretty borked!

Screenshot 2023-12-06 at 09 33 13

File: frontend/src/styles/global.scss [hunk 3]
```diff
@@ -15,7 +15,10 @@ Only 400 (`normal`), 500 (`var(--font-medium)`), 600 (`var(--font-semibold)`), o
...
+@tailwind utilities;
```
Comment (benjackwhite) [inline frontend/src/styles/global.scss:21]: FF is using some custom flex values. (This UI could probably be simplified rathe than adding the custom stuff)

Screenshot 2023-12-06 at 09 34 29

Comment (benjackwhite) [reply frontend/src/styles/global.scss:21]: Same on editing a FF - `max-w-1/2` doesn't seem to exist

Comment (benjackwhite) [reply frontend/src/styles/global.scss:21]: Same applies to new survey - I think its just the missing max width values

Comment (benjackwhite) [reply frontend/src/styles/global.scss:21]: And again for new early access feature 

Probably fixable with one off values https://tailwindcss.com/docs/min-width#arbitrary-values

**Review by benjackwhite — DISMISSED** [review]
Generally speaking this actually looks great.
Main issue is the reset header styles and some missing custom width classes

File: frontend/src/styles/global.scss (hunk 2 above)
Comment (Twixes) [reply frontend/src/styles/global.scss:19]: Fixed!

File: frontend/src/styles/global.scss (hunk 1 above)
Comment (Twixes) [reply frontend/src/styles/global.scss]: I like Tailwind's philosophy with the normalization… but we can move to it separately from the utility classes

**Comment by Twixes** [comment]
Thanks for fixing that sneaky damn Dockerfile @benjackwhite!

**Pull request merged**